src/OpenTelemetry.Exporter.InMemory/InMemoryExporterHelperExtensions.cs

```diff
@@ -27,18 +28,17 @@ public static class InMemoryExporterHelperExtensions
         /// Adds InMemory exporter to the TracerProvider.
         /// </summary>
         /// <param name="builder"><see cref="TracerProviderBuilder"/> builder to use.</param>
-        /// <param name="configure">Exporter configuration options.</param>
+        /// <param name="exportedItems">Collection to store exported telemetry.</param>
         /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "The objects should not be disposed.")]
-        public static TracerProviderBuilder AddInMemoryExporter(this TracerProviderBuilder builder, Action<InMemoryExporterOptions> configure = null)
+        public static TracerProviderBuilder AddInMemoryExporter(this TracerProviderBuilder builder, ICollection<object> exportedItems)
```

Comment:
The exporter won't function unless InMemoryExporterOptions.ExporterItems is created. By allowing an Action<InMemoryExporterOptions> as parameter, its not obvious that user *must* provide the collection themselves.

Changing to accept ICollection<object> exporterItems, forces user to pass a collection, and we can validate its non null as well.

Thoughts?

Reply:
Yeah makes sense. We want to enforce the user to pass a collection that will have the trace/logs/metrics for them to inspect and use.

Reply:
* Could we somehow expose `ICollection<Activity>` instead of `ICollection<object>`? Nicer API.

* Wondering if we should have a delegate exporter that calls an action for each Activity? I feel like that would be more useful for things other than testing. Could implement InMemoryExporter : DelegateExporter. But I have a feeling @cijothomas will say... not part of spec 😄

Reply:
Okay for both suggestions.
The background for me raising this issue is:
@utpilla was proof reading API readme.md, and realized that user will always have their activity null. (as no SDK is setup). So how do they validate that their instrumentation is correct? We are planning to add section about "How to test if my instrumentation is correct", by suggesting the use of InMemoryExporter.